test/test.js

```diff
@@ -124,39 +140,43 @@ describe('async-helpers', function() {
         };
         next(null, res);
       };
-      user.async = true;
+
       asyncHelpers.set('user', user);
       asyncHelpers.set('profile', profile);
-      var userHelper = asyncHelpers.get('user', {wrap: true});
+      var userHelper = asyncHelpers.get('user');
       var userId = userHelper('doowb');
-      assert.equal(userId, '{$ASYNCID$0$0$}');
+      assert.ok(userId.startsWith('{$ASYNCID$'));
 
-      var profileHelper = asyncHelpers.get('profile', {wrap: true});
+      var profileHelper = asyncHelpers.get('profile');
       var profileId = profileHelper(userId);
 
-      asyncHelpers.resolveIds(profileId, function(err, val) {
-        if (err) return done(err);
+      return asyncHelpers.resolve(profileId).then(function(val) {
         assert.deepEqual(val, 'doowb');
-        done();
       });
     });
   });
 
   describe('errors', function() {
-    it('should handle errors in sync helpers', function() {
+    // Not make much sense, it should be handled by the template engine
+    // We just care and handle only async helpers here.
+    // The sync helpers are processed even on engine rendering,
+    // even before the `.resolve()` calling.
+    it.skip('should handle errors in sync helpers', function() {
```

Comment:
The errors are caught and handled to add additional information to the error object. This is because not all template engines provide enough information about the error (like which helper the error occurred in).

Reply:
Kinda make sense, yea. But the only thing that is a need that I can see is to wrap the sync helper in try/catch add meta info and rethrow.